tests/ESPresense.Companion.Simulation/AccuracyHarness/Csi/ecf1_adapter.py

```
@@ -0,0 +1,201 @@
+                            ble_adverts=[(b"\x00\x11\x22\x33\x44\x55", -70, 1, 0)])
+    # ---- verify with the INDEPENDENT ecf1 container validator ----
+    h, counts, stat = ecf1.validate(out)
+    assert counts[ecf1.REC_CSI] == 10, counts
+    assert counts[ecf1.REC_BLE] == 1, counts
+    assert counts[ecf1.REC_STAT] == 1, counts
+    assert h["board_id"] == 1 and h["csi_rate_hz"] == 30 and h["n_subcarriers"] == 64, h
+    assert h["csi_payload_bytes"] == 128, h
+    # seq-gap reconstruction must agree with the metrics drop count (2)
+    assert summ["seq_gap_drop"] == 2 == summ["metrics_drop"], summ
+    # ---- byte-exact CSI: re-parse ECF1 CSI records, compare I/Q to source frames ----
+    b = open(out, "rb").read(); off = h["header_len"]; got = []
```

Ruff (0.15.15)

[error] Multiple statements on one line (semicolon)

(E702)